src/core/ext/filters/client_channel/http_proxy_mapper.cc

```diff
@@ -142,7 +142,7 @@ absl::optional<std::string> GetHttpProxyServer(
   for (size_t i = 0; i < authority_nstrs; i++) {
     gpr_free(authority_strs[i]);
   }
-  gpr_free(authority_strs);
+  gpr_free(reinterpret_cast<void*>(authority_strs));
```

Comment:
This cast shouldn't be necessary.

Reply:
I honestly have no idea why your linter is showing this as a problem.  Any pointer is automatically castable to `void*`.  I don't know of any reason why any other compiler would have a problem with this.

Reply:
Removed the cast